test/e2e/specs/wp-calypso-gutenberg-editor-tracking-spec.js

```diff
@@ -0,0 +1,95 @@
+
+import GutenbergEditorComponent from '../lib/gutenberg/gutenberg-editor-component';
+import GutenbergEditorSidebarComponent from '../lib/gutenberg/gutenberg-editor-sidebar-component';
+import SidebarComponent from '../lib/components/sidebar-component.js';
+import WPAdminSidebar from '../lib/pages/wp-admin/wp-admin-sidebar';
+
+import * as driverManager from '../lib/driver-manager.js';
+import * as dataHelper from '../lib/data-helper.js';
+import * as driverHelper from '../lib/driver-helper';
+
+const mochaTimeOut = config.get( 'mochaTimeoutMS' );
+const startBrowserTimeoutMS = config.get( 'startBrowserTimeoutMS' );
+const screenSize = driverManager.currentScreenSize();
+const host = dataHelper.getJetpackHost();
+
+const gutenbergUser =
+	process.env.GUTENBERG_EDGE === 'true' ? 'gutenbergSimpleSiteEdgeUser' : 'gutenbergSimpleSiteUser';
+
+let driver;
+
+before( async function () {
+	this.timeout( startBrowserTimeoutMS );
+	driver = await driverManager.startBrowser();
+} );
+
+describe( `[${ host }] Calypso Gutenberg Tracking: (${ screenSize })`, function () {
+	this.timeout( mochaTimeOut );
+
+	describe( 'Tracking: @parallel', function () {
+		// Create image file for upload
+		before( async function () {} );
+
+		step( 'Can log in to WPAdmin and create new Post', async function () {
+			this.loginFlow = new LoginFlow( driver, gutenbergUser );
+			if ( host !== 'WPCOM' ) {
+				this.loginFlow = new LoginFlow( driver );
+			}
+			// await this.loginFlow.loginAndStartNewPage();
+
+			await this.loginFlow.loginAndSelectWPAdmin();
+
+			//Wait for the new window or tab
+			await driver.wait( async () => ( await driver.getAllWindowHandles() ).length === 2, 10000 );
+
+			//Loop through until we find a new window handle
+			const windows = await driver.getAllWindowHandles();
+
+			await driver.switchTo().window( windows[ 1 ] );
+
+			const wpadminSidebarComponent = await WPAdminSidebar.Expect( driver );
+			await wpadminSidebarComponent.selectNewPost();
+		} );
+
+		step( 'Can track block editor events', async function () {
+			const gEditorComponent = await GutenbergEditorComponent.Expect( driver, 'wp-admin' );
+
+			// Insert some Blocks
+			await gEditorComponent.addBlock( 'Markdown' );
+			await gEditorComponent.addBlock( 'Columns' );
+			await gEditorComponent.addBlock( 'Columns' );
+
+			// Grab the events stack (only present on e2e test envs).
+			// see: https://github.com/Automattic/wp-calypso/pull/41329
+			const eventsStack = await driver.executeScript( `return window._e2eEventsStack;` );
+
+			// Assert that Insertion Events were tracked.
+			assert.strictEqual(
+				eventsStack[ 0 ][ 0 ] === 'wpcom_block_inserted',
+				true,
+				`"wpcom_block_inserted" failed for ${ eventsStack[ 0 ][ 1 ].block_name }`
+			);
+
+			assert.strictEqual(
+				eventsStack[ 0 ][ 1 ].block_name === 'core/columns',
+				true,
+				`Failed to track "block_name" property for "wpcom_block_inserted" event`
+			);
```

Comment:
I think it would be more future-proof to check through all logged events in `eventsStack` instead of checking only one item at a set index. If some other (semi- or unrelated) tracking is ever added, it might shift things up or down in the array and we will get a false negative from this test, despite all things still being tracked.

Reply:
Yep good idea. Thanks.

You will also notice that you can add `step` to break up each part of a large `describe` block. I thought maybe we should clear the `eventsStack` (from the e2e tests) after each `step` so that we know when we search the entire stack we aren't going to get false positives from stale events trigger by previous (unrelated) test. Make sense?

Reply:
Yes, the cleanup woud be required in this case. Perfect point I have missed!

Reply:
Done!